ts/DataGrid/DataGrid.ts

```diff
@@ -435,6 +453,13 @@ class DataGrid {
 
         let newDataTable = false;
         if (!this.dataTable || options.dataTable) {
+            this.userOptions.dataTable = options.dataTable;
+            if (this.options) {
+                this.options.dataTable = options.dataTable;
+            } else {
+                this.options = { dataTable: options.dataTable };
```

Comment:
```
this.userOptions.dataTable = this.options?.dataTable || {};
this.options = { dataTable: options.dataTable };
```

Moreover, the `this.userOptions.dataTable` is not used anywhere, so what is the reason of this assignment?

Reply:
Suggestion: `(this.options || {}).dataTable = options.dataTable`